network-store-iidm-impl/src/main/java/com/powsybl/network/store/iidm/impl/VariantManagerImpl.java

```diff
@@ -127,6 +128,27 @@ public void cloneVariant(String sourceVariantId, String targetVariantId, boolean
         cloneVariant(sourceVariantId, Collections.singletonList(targetVariantId), mayOverwrite);
     }
 
+    private void cloneVariantWithOptimisticConcurrency(List<VariantInfos> variantsInfos, int sourceVariantNum, String targetVariantId) {
+        List<VariantInfos> localVariantInfos = new ArrayList<>(variantsInfos);
+        boolean retry = true;
+        int attempts = 0;
+        while (retry) {
```

Comment:
```suggestion
        while (attempts < MAX_RETRY_ATTEMPTS) {
```
maybe we don't need the retry boolean if we return on success ?